### PR TITLE
[RJA-951] [fix] - lazy variables

### PR DESCRIPTION
**Pull request by rsvato**

### CODE REVIEW

File: backend/src/main/scala/com/griddynamics/genesis/template/dsl/groovy/VariableDeclaration.scala [hunk 1]
```diff
@@ -191,6 +200,17 @@ class VariableBuilder(val name : String, dsObjSupport: Option[DSObjectSupport])
             props.get(property).getOrElse(throw new MissingPropertyException(property, this.getClass))
        }
     }
+
+    override def invokeMethod(name: String, args: AnyRef) = {
+        val factory = dataSourceFactories.find(_.mode == name)
+
+        if (factory.isDefined) {
+            println("Found factory %s".format(name))
```
Comment (bugzmanov) [inline backend/src/main/scala/com/griddynamics/genesis/template/dsl/groovy/VariableDeclaration.scala]: println?

File: backend/src/main/scala/com/griddynamics/genesis/service/impl/GroovyTemplateService.scala [hunk 2]
```diff
@@ -366,7 +366,7 @@ class GroovyWorkflowDefinition(val template: EnvironmentTemplate, val workflow :
     }
 
     lazy val variableDescriptions = {
-        for (variable <- workflow.variables) yield {
+        for (variable <- workflow.variables()) yield {
             val default  = variable.defaultValue.map(String.valueOf(_)).getOrElse(null)
             val dependsOn = if (variable.dependsOn.isEmpty) None else Some(variable.dependsOn.toList)
```
Comment (ddurnev) [inline backend/src/main/scala/com/griddynamics/genesis/service/impl/GroovyTemplateService.scala:372]: Are these changes necessary: workflow.variables -> workflow.variables()?

File: backend/src/main/scala/com/griddynamics/genesis/template/dsl/groovy/VariableDeclaration.scala [hunk 3]
```diff
@@ -61,7 +62,15 @@ class VariableBuilder(val name : String, dsObjSupport: Option[DSObjectSupport])
 
     var inlineDataSource: Option[InlineDataSource] = None
 
-    var dataSourceFactories : Seq[DataSourceFactory] = Seq()
+    lazy val dsObj = {
+        val dsDelegate = new DataSourceDeclaration(projectId, dataSourceFactories)
+        val closure: Closure[Unit] = dsClosure.get
```
Comment (ddurnev) [inline backend/src/main/scala/com/griddynamics/genesis/template/dsl/groovy/VariableDeclaration.scala]: What if dsClosure is None?

File: backend/src/main/scala/com/griddynamics/genesis/service/impl/GroovyTemplateService.scala (hunk 2 above)
Comment (rsvato) [inline backend/src/main/scala/com/griddynamics/genesis/service/impl/GroovyTemplateService.scala:372]: Method with () means that some action is perfomed during call

File: backend/src/main/scala/com/griddynamics/genesis/template/dsl/groovy/VariableDeclaration.scala (hunk 3 above)
Comment (rsvato) [inline backend/src/main/scala/com/griddynamics/genesis/template/dsl/groovy/VariableDeclaration.scala]: It's not None, according to call context
